python/tests/kat/t_active_health_check.py

```diff
@@ -0,0 +1,140 @@
+import time
+from typing import Generator, Tuple, Union
+
+from abstract_tests import AmbassadorTest, HealthCheckServer, Node
+from kat.harness import Query
+
+"2022-10-04T23:24:45Z"
+"2022-10-04T23:26:45Z"
+
+
+class ActiveHealthCheckTest(AmbassadorTest):
+    def init(self):
+        self.target = HealthCheckServer()
+
+    def config(self) -> Generator[Union[str, Tuple[Node, str]], None, None]:
+        yield self, self.format(
+            """
+---
+apiVersion: getambassador.io/v3alpha1
+kind: Mapping
+name:  {self.target.path.k8s}-health
+hostname: "*"
+prefix: /healthcheck/
+service: {self.target.path.fqdn}
+resolver: endpoint
+load_balancer:
+  policy: round_robin
+health_checks:
+- unhealthy_threshold: 1
+  interval: 1s
+  health_check:
+    http:
+      path: /healthcheck/actualcheck/
+"""
+        )  # The round robin load balancer is not necessary for the test but should help make the request distribution even across the pods
+
+    def queries(self):
+        yield Query(self.url("healthcheck/"), phase=1)  # Just making sure things are running
+        yield Query(self.url("ambassador/v0/diag/"), phase=1)
+
+        yield Query(
+            self.url("healthcheck/makeUnhealthy/"), phase=1
+        )  # the deployment has 5 pods. This will make one of them start returning errors
+
+        # These two queries on their own in separate phases are just a hack way of getting the kat client
+        # to wait a little bit after the previous query so that the automated health checks have time to notice
+        # that one of the pods is misbehaving before we start blasting requests out.
+        yield Query(self.url("healthcheck/"), expected=[200, 500], phase=2)
+        yield Query(self.url("healthcheck/"), expected=[200, 500], phase=3)
+
+        # Make 1000 requests split into two groups to reduce any flakes
+        for _ in range(500):
+            yield Query(self.url("healthcheck/"), expected=[200, 500], phase=4)
+            time.sleep(0.06)
+
+        for _ in range(500):
+            yield Query(self.url("healthcheck/"), expected=[200, 500], phase=5)
+            time.sleep(0.06)
+
+    def check(self):
+        print(self.results)
+        # Check the manual requests first
+        assert self.results[0].status == 200
+        assert self.results[1].status == 200
+        assert self.results[2].status == 200
+
+        # Add up the number of 500 and 200 responses that we got.
+        valid = 0
+        errors = 0
+        for i in range(5, 1005):
+            assert (self.results[i].status == 200 or self.results[i].status == 500) == True
```

Comment:
Isn't this unecessary? the `expected` in the Query would fail the test before getting to the check function I believe.

Reply:
Ya know I think you're right. We do perform an assert on the Query for the expected result. I was just following the pattern in most of the other tests without thinking about it. Really we only need the assert on the number of 200s vs 500s so I removed em.